ws-server/src/server.rs

```diff
@@ -108,20 +109,19 @@ impl Server {
 							continue;
 						}
 
-						if connections.read().await.len() >= self.cfg.max_connections as usize {
+						if Arc::strong_count(&conn_counter) > self.cfg.max_connections as usize {
 							log::warn!("Too many connections. Try again in a while");
 							continue;
 						}
 
-						let (tx, rx) = mpsc::channel::<()>(1);
-						connections.write().await.insert(id, rx);
+						let counter = conn_counter.clone();
+						let conn = is_conn.clone();
```

Comment:
I don't have a hard rule to suggest here, but I think it's better to avoid renaming the clone to something that is not obviously a variant of the original. If I can't come up with a good name, I actually prefer numbering the clones, so `is_conn2` here.

Reply:
right, I think it could be same sense to call them `conn_counter_rc` or something but let's going with numbering for now it's better then old stuff.